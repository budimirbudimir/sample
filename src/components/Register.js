// @flow

import React, { Component } from 'react'
import { connect } from 'react-redux'

import { auth } from '../actions/user'
import '../styles/Auth.css'

type PropsFromState = {
	// registrationError is string containing error message for registration form
	registrationError?: string,
}

type PropsFromDispatch = {
	// auth is action to register new user via firebaseAuth, using email/password
	auth: (string, string) => void,
}

type Props = PropsFromState & PropsFromDispatch

class Register extends Component<Props, null> {
	email: ?HTMLInputElement
	pw: ?HTMLInputElement

	handleSubmit = e => {
		e.preventDefault()
		const { auth } = this.props

		if (this.email && this.pw) auth(this.email.value, this.pw.value)
	}

	render() {
		const { registrationError } = this.props

		return (
			<div className="Auth">
				{registrationError && (
					<p className="App-alert App-error">{registrationError}</p>
				)}

				<h3>Register</h3>

				<form onSubmit={this.handleSubmit}>
					<div className="Auth-email_container">
						<input ref={email => (this.email = email)} placeholder="Email" />
					</div>

					<div className="Auth-password_container">
						<input
							type="password"
							placeholder="Password"
							ref={pw => (this.pw = pw)}
						/>
					</div>

					{/* {this.state.registerError && <p>{this.state.registerError}</p>} */}

					<button type="submit">Register</button>
				</form>
			</div>
		)
	}
}

const mapStateToProps = state => {
	return {
		registrationError: state.user.registrationError,
	}
}

const mapDispatchToProps = dispatch => {
	return {
		auth: (email, pw) => dispatch(auth(email, pw)),
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(Register)
