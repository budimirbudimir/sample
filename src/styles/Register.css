.Register {
	padding: 15px 0;
}

.Register h3 {
	margin: 35px auto;
	font-weight: 700;
}

.Register-email_container {
	margin: 0 auto 15px;
}

.Register-password_container {
	margin: 0 auto 15px;
}

.Register input,
.Register button {
	padding: 5px 15px;
	min-width: 300px;
	max-width: 100%;
	border-radius: 4px;
}
.Register input {
	border: 1px solid #ccc;
}
.Register button {
	width: 300px;
	background: #f4425c;
	color: #fff;
	border: none;
	letter-spacing: 1px;
	cursor: pointer;
	transition: all 0.3s;
}
.Register button:hover {
	opacity: 0.5;
}
