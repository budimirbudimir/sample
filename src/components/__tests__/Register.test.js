import React from 'react'
import { shallow } from 'enzyme'

import Register from '../Register'

const registerError = ''

describe('Register', () => {
	let wrapper

	beforeEach(() => {
		wrapper = shallow(<Register regError={registerError} />)
	})

	it('matches snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('should have email input', () => {
		expect(
			wrapper.containsMatchingElement(
				<input type="text" placeholder="Email" />,
			),
		).toBe(true)
	})

	it('should have password input', () => {
		expect(
			wrapper.containsMatchingElement(
				<input type="password" placeholder="Password" />,
			),
		).toBe(true)
	})

	it('should have submit button', () => {
		expect(
			wrapper.containsMatchingElement(<button type="submit">Register</button>),
		).toBe(true)
	})
})
