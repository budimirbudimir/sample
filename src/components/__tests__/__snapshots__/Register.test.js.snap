// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    regError=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "",
        <h3>
          Register
</h3>,
        <form
          onSubmit={[Function]}
>
          <div
                    className="Auth-email_container"
          >
                    <input
                              placeholder="Email"
                              type="text"
                    />
          </div>
          <div
                    className="Auth-password_container"
          >
                    <input
                              placeholder="Password"
                              type="password"
                    />
          </div>
          <button
                    type="submit"
          >
                    Register
          </button>
</form>,
      ],
      "className": "Auth",
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Register",
        },
        "ref": null,
        "rendered": "Register",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Auth-email_container"
>
              <input
                            placeholder="Email"
                            type="text"
              />
</div>,
            <div
              className="Auth-password_container"
>
              <input
                            placeholder="Password"
                            type="password"
              />
</div>,
            <button
              type="submit"
>
              Register
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                placeholder="Email"
                type="text"
/>,
              "className": "Auth-email_container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "placeholder": "Email",
                "type": "text",
              },
              "ref": [Function],
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                placeholder="Password"
                type="password"
/>,
              "className": "Auth-password_container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "placeholder": "Password",
                "type": "password",
              },
              "ref": [Function],
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Register",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Register",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "",
          <h3>
            Register
</h3>,
          <form
            onSubmit={[Function]}
>
            <div
                        className="Auth-email_container"
            >
                        <input
                                    placeholder="Email"
                                    type="text"
                        />
            </div>
            <div
                        className="Auth-password_container"
            >
                        <input
                                    placeholder="Password"
                                    type="password"
                        />
            </div>
            <button
                        type="submit"
            >
                        Register
            </button>
</form>,
        ],
        "className": "Auth",
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Register",
          },
          "ref": null,
          "rendered": "Register",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Auth-email_container"
>
                <input
                                placeholder="Email"
                                type="text"
                />
</div>,
              <div
                className="Auth-password_container"
>
                <input
                                placeholder="Password"
                                type="password"
                />
</div>,
              <button
                type="submit"
>
                Register
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  placeholder="Email"
                  type="text"
/>,
                "className": "Auth-email_container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "placeholder": "Email",
                  "type": "text",
                },
                "ref": [Function],
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  placeholder="Password"
                  type="password"
/>,
                "className": "Auth-password_container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "placeholder": "Password",
                  "type": "password",
                },
                "ref": [Function],
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Register",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Register",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
